webapp/channels/src/components/advanced_text_editor/send_button/send_post_options/core_menu_options.tsx

```diff
@@ -23,6 +30,69 @@ function CoreMenuOptions({handleOnSelect, channelId}: Props) {
         isDM,
     } = useTimePostBoxIndicator(channelId);
 
+    const recentlyUsedCustomDate = useSelector((state: GlobalState) => getPreference(state, scheduledPosts.SCHEDULED_POSTS, scheduledPosts.RECENTLY_USED_CUSTOM_TIME));
+
+    interface RecentlyUsedCustomDate {
+        update_at?: number;
+        timestamp?: number;
+    }
+
+    let recentlyUsedCustomDateVal: RecentlyUsedCustomDate = {};
+
+    if (recentlyUsedCustomDate) {
+        try {
+            recentlyUsedCustomDateVal = JSON.parse(recentlyUsedCustomDate) as RecentlyUsedCustomDate;
+        } catch (e) {
+            recentlyUsedCustomDateVal = {};
+        }
+    }
```

Comment:
This can be moved inside useMemo to avoid recomputation on each render.

Reply:
done